### PR TITLE
Feature/search api for release calendar entry

### PR DESCRIPTION
**Pull request by StRhys**
### What

Previously florence was using babbage to get release calendar entries, but due to decommissioning the release calendar in babbage this needed to be switched to use the search api [instead](https://jira.ons.gov.uk/browse/DIS-1106). 

### How to review

- Grab this branch
- Run the search stack
- Run florence
- Login
- Check "Calendar entry schedule"
- Click "Select a calendar entry"
and if all is working, you should see release calendar entries

image
image

You can also check it against sandbox. By changing `APIRouterURL` in config to `https://api.dp.aws.onsdigital.uk` then doing `make debug` again while still logged in and refreshing. The calendar entries should now pull from sandbox.

image

* these entries won't directly match current sandbox florence, as currently babbage pulls all releases that are nott published and no cancelled, while the search api returns the not published, not cancelled _AND_ from date/time now.

### Who can review

Florence dev

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -179,6 +184,7 @@ func (svc *Service) createRouter(ctx context.Context, cfg *config.Config) (route
 
 	// API and Frontend Routers
 	router.Handle("/api/{uri:.*}", apiRouterProxy)
+	router.HandleFunc("/releasecalendar/data", ReleaseData(sc))
```
Comment (lindenmckenzie) [inline service/service.go:187]: Couldn't we just point the request at the apiRouterProxy? Or does that not work?

File: src/app/utilities/api-clients/releases.js [hunk 2]
```diff
@@ -5,19 +5,17 @@ export default class releases {
...
         }
 
         if (size) {
-            parameters.push("size=" + size);
+            parameters.push("limit=" + size);
         }
-
-        const URL = "/releasecalendar/data?view=upcoming&" + parameters.join("&");
-
+        const URL = "/releasecalendar/data?release-type=type-upcoming&" + parameters.join("&");
```
Comment (lindenmckenzie) [inline src/app/utilities/api-clients/releases.js:18]: I think retaining this URL may cause confusion.

File: service/handlers.go [hunk 3]
```diff
@@ -188,3 +190,27 @@ func DeleteHttpCookie() http.HandlerFunc {
...
+			log.Event(ctx, "error marshalling releases", log.ERROR, log.Error(err))
+			return
+		}
+
+		if _, err = w.Write(data); err != nil {
+			log.Event(ctx, "error writing response", log.ERROR, log.Error(err))
+			return
+		}
+	})
+}
```
Comment (lindenmckenzie) [inline service/handlers.go:216]: Don't think I can see any tests for this.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
A few thoughts - probably needs some further discussion, but this is a great start.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]

File: service/interfaces.go [hunk 4]
```diff
@@ -5,6 +5,7 @@ import (
 	"net/http"
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/health"
+
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
 	"github.com/ONSdigital/florence/config"
```
[the remark this answers is not in the document]
Comment (StRhys) [reply service/interfaces.go]: health and healthcheck were getting a bit close for my liking

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**